src/data/equiv/basic.lean

```diff
@@ -1625,6 +1625,15 @@ begin
   rw swap_apply_of_ne_of_ne hi hj,
 end
 
+lemma swap_inj_apply [decidable_eq β] {f : α → β} (hf : function.injective f) (x y z : α) :
```

Comment:
You could consider putting this in the `function.injective` namespace (and calling it `swap_apply`). Then we can use dot-notation on the `hf`.